src/Grpc.NetCore.HttpClient/Internal/GrpcCall.cs

```diff
@@ -0,0 +1,421 @@
+        public Task SendTask { get; private set; }
+        public HttpContentClientStreamWriter<TRequest, TResponse> ClientStreamWriter { get; private set; }
+        public HttpContextClientStreamReader<TRequest, TResponse> ClientStreamReader { get; private set; }
+
+        public GrpcCall(Method<TRequest, TResponse> method, CallOptions options, ISystemClock clock)
+        {
+            _callCts = new CancellationTokenSource();
+            Method = method;
+            Options = options;
+            _clock = clock;
+
+            if (options.CancellationToken.CanBeCanceled)
+            {
+                // The cancellation token will cancel the call CTS
+                _ctsRegistration = options.CancellationToken.Register(CancelCall);
+            }
+
+            if (options.Deadline != null && options.Deadline != DateTime.MaxValue)
+            {
+                var timeout = options.Deadline.Value - _clock.UtcNow;
+                _timeout = (timeout > TimeSpan.Zero) ? timeout : TimeSpan.Zero;
+            }
+
+            if (_timeout != null)
+            {
+                // Deadline timer will cancel the call CTS
+                _deadlineTimer = new Timer(DeadlineExceeded, null, _timeout.Value, Timeout.InfiniteTimeSpan);
+            }
+        }
+
+        public CancellationToken CancellationToken
+        {
+            get { return _callCts.Token; }
+        }
+
+        public bool IsCancellationRequested
+        {
+            get { return _callCts.IsCancellationRequested; }
+        }
+
+        public void StartUnary(System.Net.Http.HttpClient client, TRequest request)
+        {
+            var message = CreateHttpRequestMessage();
+            SetMessageContent(request, message);
+            StartSend(client, message);
+        }
+
+        public void StartClientStreaming(System.Net.Http.HttpClient client)
+        {
+            var message = CreateHttpRequestMessage();
+            ClientStreamWriter = CreateWriter(message);
+            StartSend(client, message);
+        }
+
+        public void StartServerStreaming(System.Net.Http.HttpClient client, TRequest request)
+        {
+            var message = CreateHttpRequestMessage();
+            SetMessageContent(request, message);
+            StartSend(client, message);
+            ClientStreamReader = new HttpContextClientStreamReader<TRequest, TResponse>(this);
+        }
+
+        public void StartDuplexStreaming(System.Net.Http.HttpClient client)
+        {
+            var message = CreateHttpRequestMessage();
+            ClientStreamWriter = CreateWriter(message);
+            StartSend(client, message);
+            ClientStreamReader = new HttpContextClientStreamReader<TRequest, TResponse>(this);
+        }
+
+        public void Dispose()
+        {
+            if (!Disposed)
+            {
+                Disposed = true;
+
+                _callCts.Cancel();
+                _callCts.Dispose();
+                _ctsRegistration?.Dispose();
```

Comment:
I think you need to dispose the _ctsRegistration before disposing _callCts to avoid a possible ODE in CancelCall.